agent/src/capability/baseconfigstore.h

```diff
@@ -57,41 +62,40 @@ class BaseConfigStore
      * If a capability with the same name already exists in the storage
      * the gateways will be appended to the previously stored gatway list.
      *
-     * @return ReturnCode::SUCCESS if successful
-     * @return ReturnCode::FAILURE if parsing of directory or files is unsuccessful
+     * @throws ServiceManifestParseError if parsing of the file is unsuccessful
+     * @throws ConfigStoreError if parsing of directory or files is unsuccessful
      *
      */
-    virtual ReturnCode readCapsFromDir(const std::string &dirPath);
+    virtual void readCapsFromDir(const std::string &dirPath);
 
     /**
      * @brief Reads a Service Manifest file, of type json, and adds the
      * capabilities to the internal storage.
      * If a capability with the same name already exists in the storage
      * the gateways will be appended to the previously stored gatway list.
      *
-     * @return ReturnCode::SUCCESS if successful
-     * @return ReturnCode::FAILURE if parsing of file is unsuccessful
+     * @throws ServiceManifestParseError if parsing of the file is unsuccessful
+     * @throws CapabilityParseError if parsing of one or more Capabilities in file
```

Comment:
In the readCapsFromFile it should be correct, but it was not correct for readCapsFromDir.